pkg/debugapi/pin_test.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package debugapi_test
+
+import (
+	"context"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	"github.com/ethersphere/bee/pkg/jsonhttp"
+	"github.com/ethersphere/bee/pkg/jsonhttp/jsonhttptest"
+	"github.com/ethersphere/bee/pkg/localstore"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+// TestPinChunkHandler checks for pinning, unpinning and listing of chunks.
+// It also check other edgw cases like chunk not present and checking for pinning,
+// invalid chunk address case etc. This test case has to be run in sequence and
+// it assumes some state of the DB before another case is run.
+func TestPinChunkHandler(t *testing.T) {
+	key := swarm.MustParseHexAddress("aabbcc")
+	value := []byte("data data data")
+	logger := logging.New(ioutil.Discard, 0)
+
+	storer, err := localstore.New("", key.Bytes(), nil, logger)
```

Comment:
Unit tests should test code in this package, which is the http layer and should preferably not relay on other components functionalities. A mock store should be used here instead, not the localstore, as that makes this test an integration test, which allowed new localstore pin methods not to be tested where they are implemented, in localstore package. Localstore should test for correctness of the code in it with its own tests and debugapi tests its own code.